dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -45,393 +78,519 @@ public class DidDhtConfiguration internal constructor(
+    if (types != null) {
+      // convert types to integer values
+      val typeIndexes = types.map { it.index }
+      message.addRecord(
+        TXTRecord(
+          Name("_typ._did."), DClass.IN, ttl, "id=${typeIndexes.joinToString(",")}"
+        ), Section.ANSWER
+      )
+    }
 
-                keyLookup[name.split(".")[0].drop(1)] = "$did#$vmID"
-              }
-              // handle services
-              name.startsWith("_s") -> {
-                val data = parseTxtData(rr.strings.joinToString(","))
-                services += Service.builder()
-                  .id(URI.create("$did#${data["id"]!!}"))
-                  .type(data["t"]!!)
-                  .serviceEndpoint(data["uri"]!!)
-                  .build()
-              }
-              // handle type indexing
-              name == "_typ._did." -> {
-                if (rr.strings[0].isNotEmpty() && rr.strings.size == 1) {
-                  types += rr.strings[0].removePrefix("id=").split(",").map { it.toInt() }
-                } else {
-                  throw IllegalArgumentException("invalid types record")
-                }
-              }
-              // handle root record
-              name == "_did." -> {
-                val rootData = rr.strings.joinToString(";").split(";")
-
-                val lists = mapOf(
-                  "auth" to mutableListOf<VerificationMethod>(),
-                  "asm" to mutableListOf(),
-                  "agm" to mutableListOf(),
-                  "inv" to mutableListOf(),
-                  "del" to mutableListOf()
-                )
-
-                rootData.forEach { item ->
-                  val (key, values) = item.split("=")
-                  val valueItems = values.split(",")
-
-                  valueItems.forEach {
-                    lists[key]?.add(VerificationMethod.builder().id(URI(keyLookup[it]!!)).build())
-                  }
-                }
+    return message
+  }
 
-                // add verification relationships
-                doc.authenticationVerificationMethods(lists["auth"])
-                doc.assertionMethodVerificationMethods(lists["asm"])
-                doc.keyAgreementVerificationMethods(lists["agm"])
-                doc.capabilityInvocationVerificationMethods(lists["inv"])
-                doc.capabilityDelegationVerificationMethods(lists["del"])
+  /**
+   * Converts a DNS packet to a [DIDDocument] according to the did:dht spec
+   * https://tbd54566975.github.io/did-dht-method/#dids-as-a-dns-packet
+   *
+   * @param did The DID that the packet is for.
+   * @param msg The [Message] instance containing the DNS packet.
+   * @return A [Pair] containing the [DIDDocument] and a list of types.
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   */
+  @Suppress("CyclomaticComplexMethod")
+  public fun fromDnsPacket(did: String, msg: Message): Pair<DIDDocument, List<DidDhtTypeIndexing>> {
```

Comment:
I think this should be private or internal. Doesn't seem like it should be exposed as part of the public API.

Reply:
agree will update this